### ISSUE TITLE
Bogus Warning in default interface method

### ISSUE THREAD

**Issue by GoogleCodeExporter** [typetools/checker-framework#364]
```
What steps will reproduce the problem?
1.Run the Null Checker on the class below

package test;

import java.util.Set;
import java.util.stream.Stream;

class Test {
    interface Foo {
        Set<Object> getSet();

        default void foo() {
            Stream<Object> stream = getSet().stream();
            stream.toString();
        }
    }

    static abstract class Bar {
        abstract Set<Object> getSet();

        void bar() {
            Stream<Object> stream = getSet().stream();
            stream.toString();
        }
    }
}

What is the expected output? What do you see instead?
Expected: no errors
Actual: Error at line 12: dereference of possibly-null reference stream 
stream.toString();
test\Test.java:12: error: [dereference.of.nullable] dereference of 
possibly-null reference stream
            stream.toString();
            ^
1 error

What version of the product are you using? On what operating system?
1.8.6/Windows 8/Java 8

```

Original issue reported on code.google.com by `ClovisSe...@gmail.com` on 28 Sep 2014 at 8:23
